### PR TITLE
Make the reporter interface small.

### LINKED ISSUE #9: The reporter interface has many methods
This is an issue to discuss later. 

Federico thinks the interface has a lot of methods and it would be better to define an event and only one add method that receives that event as an argument.

I think that the interface is clearer when the actions are reflected in the name of the methods, instead of defining a generic method that receives a selector argument.

### PR DESCRIPTION
**Pull request by come-maiz**
Fixes #9

### CODE REVIEW

**Comment by come-maiz** [comment]
Requires #12.

**Comment by fgimenez** [comment]
:100: !! very nice, no more empty methods :smile:

File: check.go [hunk 1]
```diff
@@ -853,31 +853,31 @@ func (runner *suiteRunner) checkFixtureArgs() bool {
 }
 
 func (runner *suiteRunner) reportCallStarted(c *C) {
-	runner.reporter.StartTest(c)
+	runner.reporter.Report(startTest, c)
```
Comment (fgimenez) [inline check.go:856]: It seems that the interface definition has not been changed, it has yet the old methods and `Report` has not been added.

File: export_test.go [hunk 2]
```diff
@@ -6,6 +6,16 @@ type TestReporter interface {
 	testReporter
 }
 
+const (
+	StartTest       = startTest
```
Comment (fgimenez) [inline export_test.go:10]: The private constants are missing.

File: check.go (hunk 1 above)
Comment (come-maiz) [inline check.go:856]: damn, I lost that change.

**Comment by come-maiz** [comment]
@fgimenez sorry about that, please take another look.

**Comment by fgimenez** [comment]
:+1: thanks

**Pull request merged**